File: week2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int countLetters(string text);
int countWords(string text);
int countSentences(string text);

int main(void)
{

    string text = get_string("Text: ");

    int letters = countLetters(text);
    int words = countWords(text);
    int sentences = countSentences(text);

    float L = (float) letters / words * 100;
    float S = (float) sentences / words * 100;

    int gradeLevel = round(0.0588 * L - 0.296 * S - 15.8);

    if (gradeLevel >= 1 && gradeLevel <= 16)
    {
        printf("Grade %i\n", gradeLevel);
    }
    else if (gradeLevel < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade 16+\n");
    }
}

int countLetters(string text)
{
    int letterCount = 0;

    for (int i = 0, length = strlen(text); i < length; i++)
    {
        if (isalpha(text[i]))
        {
            letterCount += 1;
        }
    }
    return letterCount;
}

int countWords(string text)
{
    int wordCount = 0;

    for (int i = 0, length = strlen(text); i < length; i++)
    {
        if (isblank(text[i]))
        {
            wordCount += 1;
        }
    }
    return wordCount + 1;
}

int countSentences(string text)
{
    int sentenceCount = 0;

    for (int i = 0, length = strlen(text); i < length; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentenceCount += 1;
        }
    }
    return sentenceCount;
}

File: week6_python/readability.py
from cs50 import get_string

text = get_string("Text: ")

def count_letters(text):
    letters = 0
    for char in text:
        if char.isalpha():
            letters += 1
    return letters

def count_words(text):
    words = 1
    for char in text:
        if char.isspace():
            words += 1
    return words

def count_sentences(text):
    sentences = 0
    for char in text:
        if char in '.!?':
            sentences += 1
    return sentences

def coleman_liau_index(letters, words, sentences):
    index = 0.0588 * (letters / words * 100) - 0.296 * (sentences / words * 100) - 15.8
    return round(index)

def grade_score(index):
    if index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")

letters = count_letters(text)
words = count_words(text)
sentences = count_sentences(text)
index = coleman_liau_index(letters, words, sentences)
grade_score(index)
